storage/rocksdb/ha_rocksdb.cc

```diff
@@ -3337,6 +3336,17 @@ static int rocksdb_init_func(void *const p) {
 
   DBUG_ASSERT(!mysqld_embedded);
 
+  if (rocksdb_db_options->max_open_files < 100 &&
+      rocksdb_db_options->max_open_files > 0) {
+    rocksdb_db_options->max_open_files = 100;
+  }
+
+  if (rocksdb_db_options->max_open_files > (long) open_files_limit) {
+    sql_print_information("RocksDB: rocksdb_max_open_files should not be "
+                          "greater than the open_files_limit.");
+    rocksdb_db_options->max_open_files = 100;
```

Comment:
This will not work if open_files_limit < 100

Reply:
Correct, but if open_files_limit is < 100 we probably should just abort because no reasonable instance would ever function at that level. The end result will be the same problem that initiated the bug report in the first place, an assertion down low in rocksdb because an open failed.

Reply:
I am not sure: with open_files_limit == 99 my guesstimate is that the instance would support ~30 InnoDB tables,  maybe ~45 with innodb_file_per_table=0, each arbitrarily large as long as unpartitioned. I could imagine a workable production instance with a single table too. Another surprising thing I have learned is that numerous tiny MySQL instances as backend for application correctness testing are an actually existing thing (e.g. bump buffer pool size minimum by a trivial amount, get a bug report that the test machine OOMs). In any case the code-level bug should be fixed, and if the minimum should be indeed bumped to 100 regardless of the above, that should be done as a server-level patch.

Reply:
"Tiny instance" and the use of MyRocks are somewhat mutually exclusive. 100 was just an arbitrary choice, InnoDB starts at 300 if innodb_open_files < 10, then, it sets to table_cache_size if fpt=1 or if it is > open_files_limit.

The reason that I did not just set the variable definition minimum to 100 (or whatever abstract amount) is that -1 is a valid value and used to specify 'unlimited'. Unlike InnoDB, I also did not use table_cache_size for anything as there is absolutely no relationship between the number of database.table.indexes and the number of files in the RocksDB set. I believe RocksDB would actually function with a low-ish limit but would be opening and closing SST files for nearly every operation. I think though it might be possible for compaction to require many concurrently open files.

Reply:
IMHO "tiny MyRocks instance" would still make sense in that application backend testing scenario because MyRocks is functionally different from other SEs.

Re. minimum 100 as the server-level patch, I was referring to open_files_limit. But I have not fully thought it through whether it's a good idea or not.